### ISSUE TITLE
Failing because host is a IPv6

### ISSUE THREAD

**Issue by eolexe** [davidnarayan/go-flake#1]
Hi I have found that flake.New() is failing on `getHostId()` when IP is actually IPv6

```
func getHostId() (uint64, error) {
...
    a := addrs[0].To4()
    ip := uint64((a[0] * 1 << 24) + (a[1] * 1 << 16) + (a[2] * 1 << 8) + a[3])
...
}
```

That happens because in the code above `a := addrs[0].To4()` will return nil.
What would you suggest to do in that case? 
Can you also please explain the idea of the transformations happening here? :) Thx!

**Comment by raphaeljlps** [comment]
I have the same issue,
Running OS X El Capitan

**Comment by davidnarayan** [comment]
Merged #3 to fix this issue.
